framework/src/main/java/org/tron/core/net/messagehandler/TransactionsMsgHandler.java

```diff
@@ -102,7 +103,7 @@ private void check(PeerConnection peer, TransactionsMessage msg) throws P2pExcep
   private void handleSmartContract() {
     smartContractExecutor.scheduleWithFixedDelay(() -> {
       try {
-        while (queue.size() < MAX_SMART_CONTRACT_SUBMIT_SIZE) {
+        while (queue.size() < MAX_SMART_CONTRACT_SUBMIT_SIZE && smartContractQueue.size() > 0) {
```

Comment:
Why add this logic, if the queue has no elements, it will enter an infinite loop.

Reply:
`TrxEvent event = smartContractQueue.take();`
Retrieves and removes the head of this queue, waiting if necessary until an element becomes available.
if queue is empty, will block `kill -15`.